router/batchrouter/asyncdestinationmanager/marketo-bulk-upload/marketobulkupload.go

```diff
@@ -0,0 +1,220 @@
+	var asyncResponse common.AsyncStatusResponse
+	err = json.Unmarshal(bodyBytes, &asyncResponse)
+	if err != nil {
+		panic("JSON Unmarshal Failed" + err.Error())
+	}
+	return asyncResponse, statusCode
+}
+
+func (b *MarketoBulkUploader) FetchFailedEvents(failedJobsStatus common.FetchFailedStatus) ([]byte, int) {
+	transformUrl := config.GetString("DEST_TRANSFORM_URL", "http://localhost:9090")
+	failedJobUrl := failedJobsStatus.FailedJobsURL
+	parameters := failedJobsStatus.Parameters
+	importId := gjson.GetBytes(parameters, "importId").String()
+	csvHeaders := gjson.GetBytes(parameters, "metadata.csvHeader").String()
+	payload := common.GenerateFailedPayload(b.destinationConfig, failedJobsStatus.ImportingList, importId, b.destName, csvHeaders)
+	//payload eg: "{\"config\":{\"clientId\":\"*****************\",\"clientSecret\":\"*********\",\"columnFieldsMapping\":[{\"from\":\"anonymousId\",\"to\":\"anonymousId\"},{\"from\":\"address\",\"to\":\"address\"},{\"from\":\"email\",\"to\":\"email\"}],\"deDuplicationField\":\"\",\"munchkinId\":\"585-AXP-425\",\"oneTrustCookieCategories\":[],\"uploadInterval\":\"10\"},\"input\":[{\"message\":{\"address\":23,\"anonymousId\":\"Test Kinesis\",\"email\":\"test@kinesis.com\"},\"metadata\":{\"job_id\":5}}],\"destType\":\"marketo_bulk_upload\",\"importId\":\"3095\""
+	failedBodyBytes, statusCode := misc.HTTPCallWithRetryWithTimeout(transformUrl+failedJobUrl, payload, b.timeout)
+	// failedBodyBytes eg: "{\"statusCode\":400,\"error\":\"404\"}"
+	return failedBodyBytes, statusCode
+}
+
+func (b *MarketoBulkUploader) Upload(destination *backendconfig.DestinationT, asyncDestStruct *common.AsyncDestinationStruct) common.AsyncUploadOutput {
+	resolveURL := func(base, relative string) string {
+		// var logger logger.Logger
+		baseURL, _ := url.Parse(base)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		relURL, _ := url.Parse(relative)
+		// if err != nil {
+		// 	logger.Fatal(err)
+		// }
+		destURL := baseURL.ResolveReference(relURL).String()
+		return destURL
+	}
+	destinationID := destination.ID
+	destinationUploadUrl := asyncDestStruct.URL
+	url := resolveURL(b.TransformUrl, destinationUploadUrl)
+	filePath := asyncDestStruct.FileName
+	config := destination.Config
+	destType := destination.DestinationDefinition.Name
+	failedJobIDs := asyncDestStruct.FailedJobIDs
+	importingJobIDs := asyncDestStruct.ImportingJobIDs
+
+	file, err := os.Open(filePath)
+	if err != nil {
+		panic("BRT: Read File Failed" + err.Error())
+	}
+	defer file.Close()
+	var input []common.AsyncJob
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		var tempJob common.AsyncJob
+		jobBytes := scanner.Bytes()
+		err := json.Unmarshal(jobBytes, &tempJob)
+		if err != nil {
+			panic("Unmarshalling a Single Line Failed")
+		}
+		input = append(input, tempJob)
+	}
+	var uploadT common.AsyncUploadT
+	uploadT.Input = input
+	uploadT.Config = config
+	uploadT.DestType = strings.ToLower(destType)
+	payload, err := json.Marshal(uploadT)
```

Comment:
[minor] to improve readability
```suggestion
	payload, err := json.Marshal(common.AsyncUploadT{
	        Input: input,
	        Config: config,
	        DestType: strings.ToLower(destType),
	})
```